### DISCUSSION TITLE
CMP client mode does not work properly

### DISCUSSION THREAD

**Discussion by XYQ202** [Keyfactor/ejbca-ce#57] [General]
Hello,team
      See Enrolling in **Client Mode with HMAC Password Authentication** under the **CMP Operations Guide** section of the official website
      A cmp alias is "opensslclient" is added, as shown in the figure below, and a user user2 is also added, as shown in the figure below.
     
![image](https://user-images.githubusercontent.com/97773232/177459479-f567cd41-4244-4ce9-a73a-14438693b701.png)
![image](https://user-images.githubusercontent.com/97773232/177459528-8c3bc4e3-ad4c-4da8-8038-8c6726708b57.png)

But I run the command:
     `openssl cmp -cmd ir -server localhost:8080 -path ejbca/publicweb/cmp/opensslclient -srvcert ManagementCA.cacert.pem -ref user3 -secret pass:password -certout clcert1.pem -newkey key1.pem -subject "/CN= user3/O=sunwave/C=CN"`
It does not succeed, and the following error is displayed:
        
![image](https://user-images.githubusercontent.com/97773232/177459832-1558eaa8-cc28-4194-8c11-ba8504b60b95.png)
can you help me？

Another problem is that the renewal of the certificate (KUR) cannot be successful, and an error is also reported:
CMP Error: missing protection

**Comment by primetomas** [comment]
Looks like you received an error message from the server. Check the server.log to see what was wrong with your request. It's usually in pretty clear text in the server.logl.

**Reply by XYQ202** [reply]
where is the server.logl？  I cannot find it

**Reply by Realiserad** [reply]
It's in ``/opt/wildfly/standalone/log/server.log`` if you installed EJBCA manually (on Linux) using the instructions on PrimeKey's website.

If you are using the EJBCA container with ``docker``, you can get the log output using something like ``docker logs $(docker ps | grep ejbca-ce | awk '{ print $1 }') -f``.

Anyway, I think you are missing the [``implicit-confirm`` parameter](https://www.openssl.org/docs/manmaster/man1/openssl-cmp.html).

I have [a repository with some sample commands](https://github.com/Realiserad/5g-pki-demo) which may be of help.

**Comment by Realiserad** [comment]
@XYQ202 How did it go?
